### ISSUE TITLE
Job Garage

### ISSUE THREAD

**Issue by SantyR0** [LukeWasTakenn/luke_garages#83]
Is it really a feature that you can't store your citizen car in a job garage? I mean should your car be a job vehicle? I have a business job which doesn't have a job vehicles and I'm trying to make a garage for their cars.

**Comment by LukeWasTakenn** [comment]
Yes you currently cannot store your personal vehicle in a job garage. If enough people say it shouldn't be like this then I'll change it.

**Comment by SantyR0** [comment]
How hard could that be? Is it possible for someone who is not a experienced dev like me to change it to that?

**Comment by LukeWasTakenn** [comment]
Probably not too difficult at all, this and the server side callback trigger is what controls that, so play around with it.

https://github.com/LukeWasTakenn/luke_garages/blob/8d337c8f7d8099af783276f94a65869f4d2c3abd/client/client.lua#L486-L496

**Comment by SantyR0** [comment]
Sorry, I don't understand much. I only have a little knowledge changing the codes of some scripts but this seems something I can't figure out for now. I'm currently on a phase of learning. How would I do that?

```
ESX.RegisterServerCallback('luke_garages:CheckOwnership', function(source, callback, plate, model, job)
    local xPlayer = ESX.GetPlayerFromId(source)
    local identifier = xPlayer.getIdentifier()

    plate = ESX.Math.Trim(plate)

    MySQL.Async.fetchAll('SELECT `vehicle`, `job` FROM owned_vehicles WHERE (`owner` = @owner OR `job` = @job) AND `plate` = @plate', {
        ['@owner'] = identifier,
        ['@plate'] = plate,
        ['@job'] = xPlayer.job.name
    }, function(result)
        if result[1] then
            local vehicle = json.decode(result[1].vehicle)
            local vehicleJob = result[1].job
            if ESX.Math.Trim(vehicle.plate) == plate and vehicle.model == model then
                if not job and not vehicleJob or vehicleJob == 'civ' then return callback(true) end
                if job and job == vehicleJob then return callback(true)
                else return callback({true, false}) end
            else
                -- Player tried to cheat
                callback(false)
            end
        else
            callback(false)
        end
    end)
end)
```

**Comment by SantyR0** [comment]
Hello. I was able to store a civ vehicle into my police garage but I can't get it out since I don't see it. But in my database I can see that it's on the police garage.

**Comment by LukeWasTakenn** [comment]
hmm not sure, this might be a bit more complicated than I originally thought 🤔

**Comment by TheZahamar** [comment]
I am also in favor of simply making the garage available for the job and not the car for the job

**Comment by Zzeyy** [comment]
Sure, I too think that you should be able to store your personal vehicle in a job garage.
Hell, you can even make it optional by adding a configuration for it! Please, add it.

**Comment by Zzeyy** [comment]
Man, this would be totally awesome!
Workers wouldn't have to park their car far from their job place,
instead they could park near the place, that would result in players not walking for hours.
